modules/message-lane/src/mock.rs

```diff
@@ -210,17 +213,36 @@ impl SourceHeaderChain<TestPayload, TestMessageFee> for TestSourceHeaderChain {
 #[derive(Debug)]
 pub struct TestMessageDispatch;
 
-impl MessageDispatch<TestPayload, TestMessageFee> for TestMessageDispatch {
-	fn dispatch_weight(message: &Message<TestPayload, TestMessageFee>) -> Weight {
-		message.data.payload.1
+impl MessageDispatch<TestMessageFee> for TestMessageDispatch {
+	type DispatchPayload = TestPayload;
+
+	fn dispatch_weight(message: &DispatchMessage<TestPayload, TestMessageFee>) -> Weight {
+		match message.data.payload.as_ref() {
+			Ok(payload) => payload.1,
+			Err(_) => 0,
```

Comment:
I think in a production implementation we might still want to charge for the byte-length of the payload. Without that you may send a message that you know would fail to be decoded and have the bytes included in the chain cheaply (especially if we do refunds).

Reply:
I'm not sure I understand. The `delivery_and_dispatch_fee` is paid by `submitter` in advance on the source chain. This function is called on the **target chain** when `relayer` submits deivery-transaction. The fee is already paid => bytes are already 'included' in the target chain.

Probably you mean that we should check that on the source chain instead?